### PR TITLE
The error message “Domain error in arguments” could be more informative

### PR DESCRIPTION
**Pull request by ZhihuiChen0903**
The error message “`Domain error in arguments`” seems too general to be informative.

```
C:\ProgramData\Anaconda3\lib\site-packages\scipy\stats\_distn_infrastructure.py in rvs(self, *args, **kwds)
    938         cond = logical_and(self._argcheck(*args), (scale >= 0))
    939         if not np.all(cond):
--> 940             raise ValueError("Domain error in arguments.")
    941 
    942         if np.all(scale == 0):
```

After reading the source, I believe that the expected behavior is that shape parameters should be positive and scale should >=0. Adding this information to the error message should be more user-friendly.

Expected error message:
`Domain error in arguments: shape parameters should be positive and scale should >=0.`

### CODE REVIEW

**Review by rgommers — CHANGES_REQUESTED** [review]
Thanks @ZhihuiChen0903. A better error message would be nice, but unfortunately this change is not correct - distributions define their own `_argcheck` that overrides the default one, so it's not just "must be positive". There's no easy way to get this right I'm afraid.

File: scipy/stats/_distn_infrastructure.py [hunk 1]
```diff
@@ -1030,7 +1030,7 @@ def rvs(self, *args, **kwds):
         args, loc, scale, size = self._parse_args_rvs(*args, **kwds)
         cond = logical_and(self._argcheck(*args), (scale >= 0))
         if not np.all(cond):
-            raise ValueError("Domain error in arguments.")
+            raise ValueError("Domain error in arguments:shape parameters should be positive and scale should >=0")
```
Comment (mdhaber) [inline scipy/stats/_distn_infrastructure.py:1033]:
```suggestion
            message = ("Domain error in arguments. The `scale` parameter must "
                       "be positive for all distributions; see the distribution "
                       "documentation for other restrictions.")
            raise ValueError(message)
```

Comment (mdhaber) [reply scipy/stats/_distn_infrastructure.py:1033]: Here's a thought.

Comment (mdhaber) [reply scipy/stats/_distn_infrastructure.py:1033]: @ZhihuiChen0903 would this be helpful?

**Comment by mdhaber** [comment]
No need for this to wait. I don't imagine there will be problems, but I'll re-run CI (close and re-open) and merge if it's green.

File: scipy/stats/_distn_infrastructure.py [hunk 2]
```diff
@@ -1030,7 +1030,10 @@ def rvs(self, *args, **kwds):
         args, loc, scale, size = self._parse_args_rvs(*args, **kwds)
         cond = logical_and(self._argcheck(*args), (scale >= 0))
         if not np.all(cond):
-            raise ValueError("Domain error in arguments.")
+            message = ("Domain error in arguments. The `scale` parameter must "
+                       "be positive for all distributions; see the distribution "
+                       "documentation for other restrictions.")
```
Comment (mdhaber) [inline scipy/stats/_distn_infrastructure.py:1035]:
```suggestion
            message = ("Domain error in arguments. The `scale` parameter must "
                       "be positive for all distributions; see the "
                       "distribution documentation for other restrictions.")
```

**Comment by mdhaber** [comment]
Might be helpful to use an f-string to mention the name of the distribution (e.g. "see `weibull_min` documentation"), and perhaps it should be phrased so that it's more clear that `scale` is not _necessarily_ the problem. But lint passes, and this is more informative than it was, so merging.

Thanks, @ZhihuiChen0903! Next time, please create a feature branch instead of using `master` (now `main`) for PRs, but thanks for submitting this.

**Pull request merged**